coops-ui/tests/unit/Directors.spec.ts

```diff
@@ -263,3 +265,267 @@ describe('Directors', () => {
+                  'postalCode': 'H0H 0H0',
+                  'addressRegion': 'Scotland',
+                  'deliveryInstructions': 'go to the back'
+                },
+                'mailingAddress': {
+                  'streetAddress': '1234 Street Address',
+                  'streetAddressAdditional': 'Kirkintiloch',
+                  'addressCity': 'Glasgow',
+                  'addressCountry': 'UK',
+                  'postalCode': 'H0H 0H0',
+                  'addressRegion': 'Scotland',
+                  'deliveryInstructions': 'go to the back'
+                },
+                'title': 'Treasurer'
+              }
+            ]
+          }
+      })))
+
+    const Constructor = Vue.extend(Directors)
+    const instance = new Constructor({ store: store, vuetify })
+    vm = instance.$mount()
+
+    // set as-of date
+    vm.asOfDate = '2019-04-01'
+
+    // call getDirectors() since it won't be triggered from parent component
+    vm.getDirectors()
+
+    Vue.nextTick(() => {
+      done()
+    })
+  })
+
+  afterEach(() => {
+    sinon.restore()
+  })
+
+  it('initializes the director meta data properly', () => {
+    expect(vm.directors.length).toEqual(2)
+    expect(vm.directors[0].id).toEqual(1)
+    expect(vm.directors[1].id).toEqual(2)
+  })
+
+  it('initializes the director name data properly', () => {
+    expect(vm.directors.length).toEqual(2)
+    expect(vm.directors[0].officer.firstName).toEqual('Peter')
+    expect(vm.directors[0].officer.middleInitial).toBeNull()
+    expect(vm.directors[0].officer.lastName).toEqual('Griffin')
+    expect(vm.directors[0].title).toBeNull()
+    expect(vm.directors[1].officer.firstName).toEqual('Joe')
+    expect(vm.directors[1].officer.middleInitial).toEqual('P')
+    expect(vm.directors[1].officer.lastName).toEqual('Swanson')
+    expect(vm.directors[1].title).toEqual('Treasurer')
+  })
+
+  it('initializes the director address data properly', () => {
+    // check complete first address
+    expect(vm.directors.length).toEqual(2)
+    expect(vm.directors[0].deliveryAddress.streetAddress).toEqual('mailing_address - address line one')
+    expect(vm.directors[0].deliveryAddress.streetAddressAdditional).toEqual('')
+    expect(vm.directors[0].deliveryAddress.addressCity).toEqual('mailing_address city')
+    expect(vm.directors[0].deliveryAddress.addressRegion).toEqual('BC')
+    expect(vm.directors[0].deliveryAddress.addressCountry).toEqual('mailing_address country')
+    expect(vm.directors[0].deliveryAddress.postalCode).toEqual('H0H0H0')
+    expect(vm.directors[0].deliveryAddress.deliveryInstructions).toEqual('')
+
+    // spot-check second address
+    expect(vm.directors[1].deliveryAddress.streetAddressAdditional).toEqual('Kirkintiloch')
+    expect(vm.directors[1].deliveryAddress.deliveryInstructions).toEqual('go to the back')
+
+    // Spot-check mailing address
+    expect(vm.directors[0].mailingAddress.streetAddress).toEqual('4321 Street Address')
```

Comment:
Please add a unit test where the mailing address is the same as the delivery address (and "Same as above" is displayed).

Reply:
100% will do, as per the last PR.

Reply:
This has been included in the test below ( Line 416 )